app/models/mixins/supports_feature_mixin.rb

```diff
@@ -52,14 +54,60 @@ module SupportsFeatureMixin
   #
   extend ActiveSupport::Concern
 
+  QUERYABLE_FEATURES = {
+    :associate_floating_ip    => 'Associate a Floating IP',
+    :control                  => 'Basic control operations', # FIXME: this is just a internal helper and should be refactored
+    :delete                   => 'Deletion',
+    :disassociate_floating_ip => 'Disassociate a Floating IP',
+    :discovery                => 'Discovery of Managers for a Provider',
+    :evacuate                 => 'Evacuation',
+    :events                   => 'Query for events',
+    :launch_cockpit           => 'Launch Cockpit UI',
+    :live_migrate             => 'Live Migration',
+    :migrate                  => 'Migration',
+    :provisioning             => 'Provisioning',
+    :reconfigure              => 'Reconfiguration',
+    :regions                  => 'Regions of a Provider',
+    :resize                   => 'Resizing',
+    :retire                   => 'Retirement',
+    :smartstate_analysis      => 'Smartstate Analaysis',
+  }.freeze
+
+  # Whenever this mixin is included we define all features as unsupported by default.
+  # This way we can query for every feature
+  included do
+    QUERYABLE_FEATURES.keys.each do |feature|
+      supports_not(feature)
+    end
+  end
+
+  class UnknownFeatureError < StandardError; end
+
+  def self.guard_queryable_feature(feature)
+    unless QUERYABLE_FEATURES.key?(feature.to_s.to_sym)
+      raise UnknownFeatureError, "Feature ':#{feature}' is unknown to SupportsFeatureMixin."
```

Comment:
Sorry, maybe a bit late to this.  But this raising seems to much an reaction to someone simply wants to ask.
IMHO, this guard check probably can stay with spec test but not "runtime".